pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/Utils.groovy

```diff
@@ -796,16 +796,21 @@ public class Utils {
         Set<String> previousTriggers = new HashSet<>()
         Set<String> previousParameters = new HashSet<>()
 
-        DeclarativeJobPropertyTrackerAction previousAction = null
-        WorkflowRun previousBuild = r.getPreviousBuild()
-        if (previousBuild != null) {
-            previousAction = previousBuild.getAction(DeclarativeJobPropertyTrackerAction.class)
-            if (previousAction != null) {
-                previousProperties.addAll(previousAction.getJobProperties())
-                previousTriggers.addAll(previousAction.getTriggers())
-                previousParameters.addAll(previousAction.getParameters())
+        // First, use the action from the job if it's present.
+        DeclarativeJobPropertyTrackerAction previousAction = j.getAction(DeclarativeJobPropertyTrackerAction.class)
```

Comment:
And yeah, I understand, but the thing is that this was always going to have be different to deal with `Trigger`s and `ParameterDefinition`s, so I'm not sure if we'll be able to actually use `properties` directly. I think I will end up using `JobPropertyTrackerAction.getJobPropertyDescriptors()` (and create/replace `JobPropertyTrackerAction` itself) from here, but never execute `properties`. Anyway, needs more thought.